core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -974,7 +974,7 @@ private[kafka] class Processor(
       credentialProvider.tokenCache,
       time,
       logContext,
-      () => apiVersionManager.apiVersionResponse(throttleTimeMs = 0)
+      () => apiVersionManager.apiVersionResponse(throttleTimeMs = 0, suppressFeatureLevel0 = true)
```

Comment:
why is this one always true?

Reply:
This gets used for the `ApiVersionsResponse` we send back before the SASL authentication handshake succeeds. I think this is always sent as a v0 response although I'd have to check that.

Reply:
That sounds familiar to me. Makes sense.

Reply:
Could we add a comment on why suppressFeatureLevel0 can be hard-coded to true?

Reply:
This is no longer hard-coded to true.